cosmos/providers/databricks/notebook.py

```diff
@@ -0,0 +1,199 @@
+        :param kwargs:
+    """
+
+    template_fields = ("databricks_run_id",)
+
+    def __init__(
+        self,
+        notebook_path: str,
+        source: str,
+        databricks_conn_id: str,
+        notebook_params: dict | None = None,
+        **kwargs,
+    ):
+        """
+        Create a Databricks notebook operator.
+
+        :param notebook_path: the path to the notebook in Databricks
+        :param source: Optional location type of the notebook. When set to WORKSPACE, the notebook will be retrieved
+        from the local Databricks workspace. When set to GIT, the notebook will be retrieved from a Git repository
+        defined in git_source. If the value is empty, the task will use GIT if git_source is defined
+        and WORKSPACE otherwise. For more information please visit
+        https://docs.databricks.com/dev-tools/api/latest/jobs.html#operation/JobsCreate
+        :param databricks_conn_id: the connection id to use to connect to Databricks
+        :param notebook_params: the parameters to pass to the notebook
+        :param kwargs:
+        """
+        self.notebook_path = notebook_path
+        self.source = source
+        self.notebook_params = notebook_params or {}
+        self.databricks_conn_id = databricks_conn_id
+        self.databricks_run_id = ""
+        self.job_cluster_key = kwargs.pop("job_cluster_key", "")
+        super().__init__(**kwargs)
+
+    def convert_to_databricks_workflow_task(self, relevant_upstreams):
+        """
+        Convert the operator to a Databricks workflow task that can be task in a workflow.
+
+        :type relevant_upstreams: list[BaseOperator]
+        :type job_cluster_key: dict
+        """
+        result = {
+            "task_key": self.dag_id + "__" + self.task_id.replace(".", "__"),
+            "depends_on": [
+                {"task_key": self.dag_id + "__" + t.replace(".", "__")}
+                for t in self.upstream_task_ids
+                if t in relevant_upstreams
+            ],
+            "job_cluster_key": self.job_cluster_key,
+            "timeout_seconds": 0,
+            "email_notifications": {},
+            "notebook_task": {
+                "notebook_path": self.notebook_path,
+                "source": self.source,
+                "base_parameters": self.notebook_params,
+            },
+        }
+        return result
+
+    def monitor_databricks_job(self):
+        """Monitor the Databricks job until it completes. Raises Airflow exception if the job fails."""
+        hook = DatabricksHook(self.databricks_conn_id)
+        databricks_conn = hook.get_conn()
+        api_client = ApiClient(
+            host=databricks_conn.host,
+            token=databricks_conn.password,
+        )
+        runs_api = RunsApi(api_client)
+        current_task = {
+            x["task_key"]: x for x in runs_api.get_run(self.databricks_run_id)["tasks"]
+        }[self.dag_id + "__" + self.task_id.replace(".", "__")]
+        while (
```

Comment:
Should we set some timeout here?

Reply:
We agreed to do this as a separate ticket - @dimberman to help log this ticket and adding the reference here

Reply:
https://github.com/astronomer/astronomer-cosmos/issues/145 linked here